staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -293,12 +293,14 @@ func (s *WatchServer) HandleWS(ws *websocket.Conn) {
 		case event, ok := <-ch:
 			if !ok {
 				// End of results.
+				s.Watching.Stop()
```

Comment:
I suspect this is the only case in which s.Watching.Stop() shouldn't be called -- it's already stopped.

If I am not correct, or if it is safe to double-call Stop(), then s.Watching.Stop() should be called via defer instead of by every return.

Reply:
I will drop this change and focus on the the other change.